k9mail/src/main/java/com/fsck/k9/view/MessageCryptoDisplayStatus.java

```diff
@@ -237,12 +252,15 @@ private static MessageCryptoDisplayStatus getDisplayStatusForPgpResult(CryptoRes
             }
         }
 
+        final long delta = signatureResult.getSignatureTimestamp().getTime() - messageSentDate.getTime();
```

Comment:
I also thought about if this would be better put this into OKC. i'm also using it for XMPP. BUT I also use it for a password-app and for a simple plain-text-sign-decrypt-app, and there I don't need it. So I came to the conclusion, that the feature might be more In the email context.

If you are talking about "Conversations" jabber client: that encrypts/signs the whole message including timestamp. Having the timestamp in the header and the message in the "body" - and apply the signature/encryption only to the "body" is actual more an email-problem.

Regarding messages without Date: I didn't consider this. But wouldn't that cause much more issues, (for example) since all emails are sorted by date?